WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java

```diff
@@ -205,10 +205,21 @@ private void unpackMediaIntent(@NonNull Intent intent) {
             // if this media belongs to some post, register such Post
             registerPostModelsForMedia(mediaList, intent.getBooleanExtra(KEY_SHOULD_RETRY, false));
 
+            ArrayList<MediaModel> toBeUploadedMediaList = new ArrayList<>(mediaList);
```

Comment:
I'm wondering if we could improve clarify of this code. There are some things I found a bit confusing while parsing it:
- "toBeUploadedMediaList" is initialized with media we might not want to upload
- "toBeUploadedMediaList" is not used to decide what we want to upload, it's used only to show the notification

I'd consider the following:
- initialize "toBeUploadedMediaList" as an empty list
- if the media was already uploaded, don't even add it to the list
- iterate over "toBeUploadedMediaList" and enqueue all items from this list

Something like this:
```
ArrayList<MediaModel> toBeUploadedMediaList = new ArrayList<>();
            for (MediaModel media : mediaList) {
                MediaModel localMedia = mMediaStore.getMediaWithLocalId(media.getId());
                boolean notUploadedYet = localMedia != null
                                               && MediaUploadState.fromString(localMedia.getUploadState())
                                                  != MediaUploadState.UPLOADED;
                if (notUploadedYet) {
                    toBeUploadedMediaList.add(localMedia);
                }
            }

            for (MediaModel media : toBeUploadedMediaList) {
                mMediaUploadHandler.upload(media);
            }

            if (!toBeUploadedMediaList.isEmpty()) {
                mPostUploadNotifier.addMediaInfoToForegroundNotification(toBeUploadedMediaList);
            }
```
Wdyt?

Reply:
Makes sense :)! Done in b78ebcc.